tests/foreman/cli/test_environment.py

```diff
@@ -83,6 +83,52 @@ def test_list(self):
         self.assertTrue(len(result.stdout) == 1,
                         "Environment list - stdout contains 'Name'")
 
+    def test_create_environment_with_location(self):
+        """@Test: Check if Environment with Location can be created
+
+        @Feature: Environment - Create
+
+        @Assert: Environment is created and has new Location assigned
+
+        """
+        name = gen_string("alpha", 10)
+        try:
+            new_loc = make_location()
+            make_environment({
+                'name': name,
+                'location-ids': new_loc["id"],
+            })
+        except CLIFactoryError as err:
+            self.fail(err)
+
+        result = Environment().info({'name': name})
```

Comment:
This info is not necessary, the factory already run info for you. You can drop lines 104-106 and refactor 107 to:

```python
self.assertIn(new_loc['name'], environment['locations'])
```